invoice.go

```diff
@@ -107,6 +101,12 @@ type SalesItemLineDetail struct {
 	DiscountAmt     json.Number   `json:",omitempty"`
 }
 
+// DiscountLineDetail ...
+type DiscountLineDetail struct {
+	PercentBased    bool
+	DiscountPercent float32 `json:",omitempty"`
+}
```

Comment:
How would it handle flat discount amounts? Isn't there another property you need here?

Reply:
Flat discount amounts seem to work by setting the `Amount` property on the `Line`, with `DiscountPercent` unset and `PercentBased` as false. If I remember correctly it needed the `DiscountLineDetail` to not be missing though, which is why I didn't put `omitempty` on `PercentBased` (since the `DiscountLineDetail` itself on the `Line` is `omitempty`)